saleor/webhook/payloads.py

```diff
@@ -996,3 +999,83 @@ def generate_translation_payload(
         translation_data.update(context)
 
     return json.dumps(translation_data)
+
+
+MAX_PAYLOAD_ATTRIBUTE_SIZE = 1024 * 125
+
+
+def truncate_str_to_byte_limit(
+    text: Optional[str], limit=MAX_PAYLOAD_ATTRIBUTE_SIZE, encoding="utf-8"
+) -> Optional[str]:
+    if not text:
+        return text
+    return text.encode(encoding)[:limit].decode(encoding, "ignore")
+
+
+def generate_api_call_payload(request, response):
+    content_length = int(request.headers.get("Content-Length", 0))
+    response_body = response.content.decode(response.charset)
+    request_body = None
+    if request.POST:
+        request_body = json.dumps(dict(request.POST))
+    else:
+        try:
+            request_body = request.body.decode("utf-8")
+        except ValueError:
+            pass
+    request_id = None
+    if hasattr(request, "request_uuid") and request.request_uuid:
+        request_id = str(request.request_uuid)
+
+    payload = {
+        "request_id": request_id or str(uuid.uuid4()),
```

Comment:
What exactly is achieved by adding `request_id`?  I mean, assuming that someone will come to you as an admin, and will send you an ID of a failed request. We don't track it anywhere on our side.

Reply:
Because of transport channels, the report receiver can receive the same report multiple times. `request_id` is added to identify them on the receiver side and skip duplicates.